source/opt/propagator.cpp

```diff
@@ -36,22 +36,14 @@ void SSAPropagator::AddControlEdge(const Edge& edge) {
   blocks_.push(dest_bb);
 }
 
-void SSAPropagator::AddSSAEdges(ir::Instruction* instr, bool traverse_phis) {
+void SSAPropagator::AddSSAEdges(ir::Instruction* instr) {
   // Ignore instructions that produce no result.
   if (instr->result_id() == 0) {
     return;
   }
 
   get_def_use_mgr()->ForEachUser(
-      instr->result_id(), [this, traverse_phis](ir::Instruction* use_instr) {
-        // If |use_instr| is a Phi, ignore this edge.  Phi instructions can form
-        // cycles in the def-use web, which would get the propagator into an
```

Comment:
Oh, nice I always disliked this hack. However, how did your change guarantee that we don't get into an infinite loop anymore?

Reply:
I tried both infinite loop cases linked through #1159 and neither hangs.

Reply:
You will not get into an infinite loop because you only add the ssa edges if the status changed.  For each instruction, it status can change at most twice (not interesting -> interesting -> varying).  Then the simulations will stop.